spdx-utils/src/main/kotlin/model/SpdxFile.kt

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright (C) 2017-2020 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.spdx.model
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties
+import com.fasterxml.jackson.annotation.JsonInclude
+import com.fasterxml.jackson.annotation.JsonProperty
+
+import org.ossreviewtoolkit.spdx.isSpdxExpressionOrNotPresent
+
+/**
+ * Provides important meta-data about a particular file of a software package.
+ */
+@JsonIgnoreProperties("ranges") // TODO: Implement ranges which is broken in the specification examples.
+data class SpdxFile(
+    /**
+     * A unique identifies this [SpdxFile] within a SPDX document.
+     */
+    @JsonProperty("SPDXID")
+    val spdxId: String,
+
+    /**
+     * The [SpdxAnnotation]s for the file.
+     */
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    val annotations: List<SpdxAnnotation> = emptyList(),
+
+    /**
+     * Checksums of the file, must contain at least one entry using [SpdxChecksum.Algorithm.SHA1].
+     */
+    val checksums: List<SpdxChecksum>,
+
+    /**
+     * A general comment about the file.
+     */
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    val comment: String = "",
+
+    /**
+     * A text relating to a copyright notice, even if not complete. To represent a not present value
+     * [SpdxConstants.NONE] or [SpdxConstants.NOASSERTION] must be used.
+     */
+    val copyrightText: String,
+
+    /**
+     * The list of contributors which contributed to the file. Contributors could include names of copyright holders
+     * and/or authors who may not be copyright holders, yet contributed to the file content.
+     */
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
+    val fileContributors: List<String>,
```

Comment:
I believe this should either be an empty list by default, or the annotation should go away.

Reply:
Done.